src/Azure.Functions.Cli/Actions/BaseAction.cs

```diff
@@ -27,6 +42,23 @@ public void SetFlag<T>(string longOption, string description, Action<T> callback
             }
         }
 
+        public virtual void UpdateTelemetryEvent(TelemetryEvent telemetryEvent)
```

Comment:
This should likely be moved to `TelemetryHelpers` and take the events as a parameter.

Reply:
I think I created this virtual method with the idea that any Actions can override and manipulate the `telemetryEvent`. But then I moved to having a local `TelemetryCommandEvents` for commands to write to. I still left this method here, as -- 

a) it does have a dependency on the local `TelemetryCommandEvents` (which I think I should have made private), and 
b) still allows commands to override this and manipulate the telemetryevent, as this is latery called by the ConsoleApp block to update the TelemetryEvent.


If we think that's a better design, I can move it to `TelemetryHelpers`, and call that from ConsoleApp, would just need to leave `TelemetryCommandEvents` with a public getter.

Reply:
Unless there's a strong reason you can see to keep this here, I would recommend moving this to the helper as it is out of place in this type (this logic doesn't really belong here). Passing the events as an argument to this would be a better option.

Reply:
Yeah, thinking about it, you are right. It doesn't belong here. Nothing specific to the action is happening in the function. I will update this.